users/policies/service_test.go

```diff
@@ -154,23 +158,27 @@ func TestAddPolicy(t *testing.T) {
 
 	for _, tc := range cases {
 		repoCall := pRepo.On("CheckAdmin", context.Background(), mock.Anything).Return(nil)
-		repoCall1 := pRepo.On("Evaluate", context.Background(), "client", mock.Anything).Return(nil)
-		repoCall2 := pRepo.On("Update", context.Background(), tc.policy).Return(tc.err)
+		repoCall1 := pRepo.On("Retrieve", context.Background(), mock.Anything).Return(tc.page, nil)
```

Comment:
Retrieve, Save, and Update make sense, and so does `CheckAdmin`. However, `Authorize is not called here. Can you point out where the `CheckAdmin` call comes from? Also, can that one be embedded in Evaluate policy call?

Reply:
`Authorize` on service layer contains both `CheckAdmin` which comes before `Evaluate`. We are using a token from an admin user. If it would have been a normal user we would have both `CheckAdmin` and `Evaluate`